docs/conf.py

```diff
@@ -35,8 +35,16 @@
     "sphinx_copybutton",
     "sphinx.ext.doctest",
     "sphinx.ext.intersphinx",
+    "sphinxext.opengraph",
 ]
 
+# Open Graph metadata
+ogp_title = "urllib3 documentation"
+ogp_site_url = "https://urllib3.readthedocs.io"
+ogp_type = "website"
+ogp_image = "https://github.com/urllib3/urllib3/raw/main/docs/_static/banner_github.svg"
```

Comment:
OpenGraph does not support SVG, use a PNG instead:

https://stackoverflow.com/a/26876671/724176

Compare https://github.com/python-pillow/Pillow/pull/5498

Reply:
I found that with the following change, it generates social cards for every page
```diff
diff --git a/docs/conf.py b/docs/conf.py
index 69f5c4dd..a2a93fdb 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -42,7 +42,7 @@ extensions = [
 ogp_title = "urllib3 documentation"
 ogp_site_url = "https://urllib3.readthedocs.io"
 ogp_type = "website"
-ogp_image = "https://github.com/urllib3/urllib3/raw/main/docs/_static/banner_github.svg"
+ogp_social_cards = {"image": "images/logo.png", "line_color": "#F09837"}
 ogp_description = "urllib3 is a user-friendly HTTP client library for Python."
 
 # Test code blocks only when explicitly specified
```

![](https://github.com/urllib3/urllib3/assets/17710133/9c8e7600-6b40-42be-9227-58f33233fd0c)

@sethmlarson @pquentin what do you this about using them instead of a static image?

FYI, the little icon is Sphinx's

Reply:
Sounds good to me, the logo is enough here, `urllib3` is already mentioned multiple times.